File: SampleApp/java/app/src/main/java/jp/fluct/sample/samplefluctsdkapp/RewardedVideoActivity.java
package jp.fluct.sample.samplefluctsdkapp;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import java.util.Calendar;
import java.util.GregorianCalendar;
import jp.fluct.fluctsdk.FluctAdRequestTargeting;
import jp.fluct.fluctsdk.FluctErrorCode;
import jp.fluct.fluctsdk.FluctRewardedVideo;
import jp.fluct.fluctsdk.FluctRewardedVideoSettings;

public class RewardedVideoActivity extends AppCompatActivity {

    private FluctRewardedVideo rewardedVideo;
    private Button showButton;
    private TextView stateTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rewarded_video);

        stateTextView = (TextView) findViewById(R.id.state_textview);

        String groupID = "1000090271";
        String unitID = "1000135434";

        FluctRewardedVideoSettings settings = new FluctRewardedVideoSettings.Builder()
            .testMode(true)
            .debugMode(true)
            .build();

        rewardedVideo = FluctRewardedVideo.getInstance(groupID, unitID, this, settings);

        rewardedVideo.setListener(
            new FluctRewardedVideo.Listener() {
                // 広告読み込み完了
                @Override
                public void onLoaded(String groupId, String unitId) {
                    updateStateText("Loaded");
                    showButton.setEnabled(true);
                    Log.d("RewardedVideoActivity", "onLoaded");
                }

                // 広告読み込み失敗
                @Override
                public void onFailedToLoad(
                    String groupId,
                    String unitId,
                    FluctErrorCode errorCode
                ) {
                    updateStateText("Failed to load");
                    Log.d("RewardedVideoActivity", "onFailedToLoad");
                }

                // 広告が表示した
                @Override
                public void onOpened(String groupId, String unitId) {
                    Log.d("RewardedVideoActivity", "onOpened");
                }

                // 動画が再生された
                @Override
                public void onStarted(String groupId, String unitId) {
                    Log.d("RewardedVideoActivity", "onStarted");
                }

                // リワード付与通知
                @Override
                public void onShouldReward(String groupId, String unitId) {
                    Log.d("RewardedVideoActivity", "onShouldReward");
                }

                // 広告が閉じられた
                @Override
                public void onClosed(String groupId, String unitId) {
                    updateStateText("Complete");
                    showButton.setEnabled(false);
                    Log.d("RewardedVideoActivity", "onClosed");
                }

                // 広告の再生が失敗した
                @Override
                public void onFailedToPlay(
                    String groupId,
                    String unitId,
                    FluctErrorCode errorCode
                ) {
                    updateStateText("Failed to play");
                    Log.d("RewardedVideoActivity", "onFailedToPlay");
                }
            }
        );

        showButton = (Button) findViewById(R.id.show_button);
        showButton.setOnClickListener(
            new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (!rewardedVideo.isAdLoaded()) {
                        return;
                    }

                    rewardedVideo.show();
                }
            }
        );

        findViewById(R.id.load_button)
            .setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        FluctAdRequestTargeting targeting = new FluctAdRequestTargeting();
                        targeting.setGender(FluctAdRequestTargeting.FluctGender.MALE);
                        targeting.setBirthday(
                            new GregorianCalendar(1988, Calendar.JANUARY, 1).getTime()
                        );
                        rewardedVideo.loadAd(targeting);
                        updateStateText("Loading");
                    }
                }
            );
    }

    @SuppressLint("SetTextI18n")
    private void updateStateText(String s) {
        stateTextView.setText("State: " + s);
    }
}

File: SampleApp/kotlin/app/src/main/java/jp/fluct/sample/kotlinapp/RewardedVideoActivity.kt
package jp.fluct.sample.kotlinapp

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import jp.fluct.fluctsdk.FluctAdRequestTargeting
import jp.fluct.fluctsdk.FluctErrorCode
import jp.fluct.fluctsdk.FluctRewardedVideo
import jp.fluct.fluctsdk.FluctRewardedVideoSettings
import java.util.*

class RewardedVideoActivity : AppCompatActivity() {

    companion object {

        private const val GROUP_ID = "1000090271"
        private const val UNIT_ID = "1000135434"

        private const val SAVED_STATE_UI_STATE_TEXT = "ui_state_text"
        private const val SAVED_STATE_UI_LOAD_BUTTON_ENABLED = "ui_load_button_enabled"
        private const val SAVED_STATE_UI_SHOW_BUTTON_ENABLED = "ui_show_button_enabled"

    }

    private lateinit var stateTextView: TextView
    private lateinit var showButton: Button
    private lateinit var loadButton: Button

    private fun getRewardedVideoInstance(): FluctRewardedVideo {
        val settings = FluctRewardedVideoSettings.Builder()
            .testMode(true)
            .debugMode(true)
            .build()
        return FluctRewardedVideo.getInstance(
            GROUP_ID,
            UNIT_ID,
            this,
            settings
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rewardedvideo)

        stateTextView = findViewById<TextView>(R.id.state_textview)
        showButton = findViewById<Button>(R.id.show_button)
        loadButton = findViewById<Button>(R.id.load_button)

        showButton.setOnClickListener {
            if (getRewardedVideoInstance().isAdLoaded) {
                getRewardedVideoInstance().show()
            }
        }
        loadButton.setOnClickListener {
            if (!getRewardedVideoInstance().isAdLoaded) {
                val targeting = FluctAdRequestTargeting()
                targeting.gender = FluctAdRequestTargeting.FluctGender.MALE
                targeting.birthday = GregorianCalendar(1988, Calendar.JANUARY, 1).time

                getRewardedVideoInstance().loadAd(targeting)

                updateUi(
                    state = "State: Loading",
                    loadButtonEnabled = false
                )
            }
        }
    }

    override fun onResume() {
        super.onResume()
        getRewardedVideoInstance().setListener(object : FluctRewardedVideo.Listener {
            // 広告読み込み完了
            override fun onLoaded(groupId: String?, unitId: String?) {
                updateUi(
                    state = "State: Loaded",
                    loadButtonEnabled = false,
                    showButtonEnabled = true
                )
                Log.d("RewardedVideoActivity", "onLoaded")
            }

            // 広告読み込み失敗
            override fun onFailedToLoad(groupId: String?, unitId: String?, errorCode: FluctErrorCode?) {
                updateUi(
                    state = "State: Failed to load",
                    loadButtonEnabled = true
                )
                Log.d("RewardedVideoActivity", "onFailedToLoad")
            }

            // 広告が表示した
            override fun onOpened(groupId: String?, unitId: String?) {
                Log.d("RewardedVideoActivity", "onOpened")
            }

            // 動画が再生された
            override fun onStarted(groupId: String?, unitId: String?) {
                Log.d("RewardedVideoActivity", "onStarted")
            }

            // リワード付与通知
            override fun onShouldReward(groupId: String?, unitId: String?) {
                Log.d("RewardedVideoActivity", "onShouldReward")
            }

            // 広告が閉じられた
            override fun onClosed(groupId: String?, unitId: String?) {
                updateUi(
                    state = "State: Complete",
                    loadButtonEnabled = true,
                    showButtonEnabled = false
                )
                Log.d("RewardedVideoActivity", "onClosed")
            }

            // 広告の再生が失敗した
            override fun onFailedToPlay(groupId: String?, unitId: String?, errorCode: FluctErrorCode?) {
                Log.d("RewardedVideoActivity", "onFailedToPlay")
            }
        })
    }

    @SuppressLint("SetTextI18n")
    private fun updateUi(
        state: String,
        loadButtonEnabled: Boolean = loadButton.isEnabled,
        showButtonEnabled: Boolean = showButton.isEnabled
    ) {
        stateTextView.text = state
        loadButton.isEnabled = loadButtonEnabled
        showButton.isEnabled = showButtonEnabled
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString(SAVED_STATE_UI_STATE_TEXT, stateTextView.text.toString())
        outState.putBoolean(SAVED_STATE_UI_LOAD_BUTTON_ENABLED, loadButton.isEnabled)
        outState.putBoolean(SAVED_STATE_UI_SHOW_BUTTON_ENABLED, showButton.isEnabled)
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        updateUi(
            state = savedInstanceState.getString(SAVED_STATE_UI_STATE_TEXT) ?: "",
            loadButtonEnabled = savedInstanceState.getBoolean(SAVED_STATE_UI_LOAD_BUTTON_ENABLED),
            showButtonEnabled = savedInstanceState.getBoolean(SAVED_STATE_UI_SHOW_BUTTON_ENABLED)
        )
    }
}
